internal/planner/planner.go

```diff
@@ -286,41 +286,31 @@ func getVersionConfigDiff(
 	strategy temporaliov1alpha1.RolloutStrategy,
 	status *temporaliov1alpha1.TemporalWorkerDeploymentStatus,
 	conflictToken []byte,
-) *VersionConfig {
-	vcfg := getVersionConfig(l, strategy, status)
-	if vcfg == nil {
-		return nil
-	}
-
-	// Check if target version is nil
-	if status == nil || status.TargetVersion == nil {
-		return nil
-	}
-
-	vcfg.VersionID = status.TargetVersion.VersionID
-	vcfg.ConflictToken = conflictToken
+) []*VersionConfig {
+	var versionConfigs []*VersionConfig
 
-	// Set default version if there isn't one yet
-	if status.DefaultVersion == nil {
-		vcfg.SetDefault = true
-		vcfg.RampPercentage = 0
-		return vcfg
-	}
+	vcfg := getVersionConfig(l, strategy, status)
+	if vcfg != nil {
+		vcfg.VersionID = status.TargetVersion.VersionID
+		vcfg.ConflictToken = conflictToken
 
-	// Don't make updates if desired default is already the default
-	if vcfg.VersionID == status.DefaultVersion.VersionID {
-		return nil
+		versionConfigs = append(versionConfigs, vcfg)
 	}
 
-	// Don't make updates if desired ramping version is already the target, and ramp percentage is correct
-	if !vcfg.SetDefault &&
-		vcfg.VersionID == status.TargetVersion.VersionID &&
-		status.TargetVersion.RampPercentage != nil &&
-		vcfg.RampPercentage == *status.TargetVersion.RampPercentage {
-		return nil
+	for _, vcfg := range status.DeprecatedVersions {
```

Comment:
Lets say version A is current, and then version B is the target version and starts rolling out gradually.
Then someone notices a problem with B so they make A the new target (let's say with Immediate cutover strategy).

We want the ramp of B to be set to 0 / the ramp to be "unset".
It seems like that would only happen if B is in status.DeprecatedVersions. Is that the case? (The contents of `DeprecatedVersions` may have changed since I last read the code, or I forgot)

Reply:
The ramping version would indeed have been in DeprecatedVersions, but my assumption was that it was possible for multiple of those to have a ramp set for some reason. I now understand that is not the case, so I've being more explicit and using RampingVersion as your PR did.